src/main/java/net/datafaker/idnumbers/NorwegianIdNumber.java

```diff
@@ -0,0 +1,120 @@
+ *     <li>yy - year of birth</li>
+ *     <li>CC - sequence (range depends on century)</li>
+ *     <li>kk - checksum</li>
+ * </ul>
+ * <p>
+ *     Example: 11077941012
+ * </p>
+ *
+ * @see <a href="https://www.skatteetaten.no/en/person/national-registry/identitetsnummer-og-elektronisk-id/fodselsnummer/">Overview</a>
+ * @see <a href="https://taxid.pro/">Online generator</a>
+ * @see <a href="https://www.oecd.org/content/dam/oecd/en/topics/policy-issue-focus/aeoi/norway-tin.pdf">More on centuries</a>
+ */
+public class NorwegianIdNumber implements IdNumberGenerator {
+    private static final DateTimeFormatter BIRTHDAY_FORMAT = DateTimeFormatter.ofPattern("ddMMyy");
+
+    private static final int[] CHECKSUM_COEFFICIENTS_K1 = {3, 7, 6, 1, 8, 9, 4, 5, 2};
+    private static final int[] CHECKSUM_COEFFICIENTS_K2 = {5, 4, 3, 2, 7, 6, 5, 4, 3, 2};
+
+    @Override
+    public String countryCode() {
+        return "NO";
+    }
+
+    @Override
+    public PersonIdNumber generateValid(BaseProviders faker, IdNumberRequest request) {
+        LocalDate birthday = birthday(faker, request);
+        Gender gender = gender(faker, request);
+
+        String basePart = basePart(faker, birthday, gender);
+        String idNumber = "%s%02d".formatted(basePart, checksum(basePart));
+        return new PersonIdNumber(idNumber, birthday, gender);
+    }
+
+    @Override
+    public String generateInvalid(BaseProviders faker) {
+        String valid = generateValid(faker);
+        String basePart = valid.substring(0, valid.length() - 2);
+        int invalidChecksum = (checksum(basePart) + 1) % 100;
+        return "%s%02d".formatted(basePart, invalidChecksum);
+    }
+
+    String basePart(BaseProviders faker, LocalDate birthday, Gender gender) {
+        String birthdayDigits = BIRTHDAY_FORMAT.format(birthday);
+        int sequenceNumber = generateSequenceNumber(faker, birthday.getYear());
+        int genderDigit = genderDigit(faker, gender);
+
+        return "%s%02d%s".formatted(birthdayDigits, sequenceNumber, genderDigit);
+    }
+
+    /**
+     * <ul>
+     * <li>born 1854-1899: allocated from series 749-500</li>
+     * <li>born 1900-1999: allocated from series 499-000</li>
+     * <li>born 1940-1999: also allocated from series 999-900</li>
+     * <li>born 2000-2039: allocated from series 999-500</li>
+     * </ul>
+     *
+     * @see <a href="https://www.oecd.org/content/dam/oecd/en/topics/policy-issue-focus/aeoi/norway-tin.pdf">TIN Structure</a>
+     */
+    private int generateSequenceNumber(BaseProviders faker, int year) {
+        if (year >= 1854 && year <= 1899) {
+            return faker.random().nextInt(50, 74);
+        } else if (year >= 1940 && year <= 1999) {
+            return faker.random().nextInt(90, 99);
+        } else if (year >= 1900 && year <= 1999) {
+            return faker.random().nextInt(0, 49);
+        } else if (year >= 2000 && year <= 2039) {
+            return faker.random().nextInt(50, 99);
```

Comment:
can you elaborate why javadoc and the doc (https://www.oecd.org/content/dam/oecd/en/topics/policy-issue-focus/aeoi/norway-tin.pdf) says
>The three digits comprising the individual number are allocated sequentially within the
specific date of birth. Individual digits are allocated as follows

and this method returns 1 or 2 digits?